modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java

```diff
@@ -86,13 +88,29 @@ public void destroy() {
     @Override
     public void addSession(Session session) {
 
-        activeDebugSessions.put(session, createSession(session));
+        if (getTenantDomain(session) != null && !activeDebugSessions.containsKey(getTenantDomain(session))) {
+            activeDebugSessions.put(getTenantDomain(session), createSession(session));
+
+        } else {
+            try {
+                JsonDap jsonDap = new JsonDap();
+                jsonDap.init();
+                ProtocolMessage message = new ProtocolMessage(DAPConstants.DEBUG_ERROR_DEBUG_SESSION_ALIVE);
+                String text = jsonDap.encode(message);
+                session.getBasicRemote().sendText(text);
+
+            } catch (IOException ex) {
+                log.error("Error on Encoding the message", ex);
+
+            }
+
```

Comment:
Let's remove the additional newline here.

Reply:
Fixed with https://github.com/wso2-extensions/identity-tools-debugger/pull/17